creator/lib/app.dart

```diff
@@ -1,58 +1,142 @@
+            (Set<MaterialState> states) {
+              if (states.contains(MaterialState.disabled)) {
+                return Colors.white38;
+              }
+              return Colors.blue;
+            },
           ),
         ),
+      ),
+      scaffoldBackgroundColor: primaryColor,
+      appBarTheme: AppBarTheme(
+        backgroundColor: Colors.white,
         textTheme: TextTheme(
           headline1: TextStyle(
             color: primaryColor,
-            fontSize: 32,
-            fontWeight: FontWeight.bold,
-          ),
-          headline2: TextStyle(
-            color: Colors.white,
-            fontSize: 24,
-            fontWeight: FontWeight.w600,
-          ),
-          headline3: TextStyle(
-            color: primaryColor,
-            fontSize: 16,
-          ),
-          bodyText1: TextStyle(
-            color: Colors.white,
-            fontSize: 16,
-            fontWeight: FontWeight.w300,
           ),
         ),
       ),
+      textTheme: TextTheme(
+        headline1: TextStyle(
+          color: primaryColor,
+          fontSize: 32,
+          fontWeight: FontWeight.bold,
+        ),
+        headline2: TextStyle(
+          color: Colors.white,
+          fontSize: 24,
+          fontWeight: FontWeight.w600,
+        ),
+        headline3: TextStyle(
+          color: primaryColor,
+          fontSize: 16,
+        ),
+        bodyText1: TextStyle(
+          color: Colors.white,
+          fontSize: 16,
+          fontWeight: FontWeight.w300,
+        ),
+      ),
+    );
+  }
+
+  TextTheme _buildTextTheme(ThemeData defaultThemeData) =>
+      defaultThemeData.textTheme
+          .copyWith(
+            headline1: TextStyle(
+              fontSize: 32,
+              fontWeight: FontWeight.bold,
+            ),
+            headline3: TextStyle(
+              color: primaryColor,
+              fontSize: 20,
+              fontWeight: FontWeight.w700,
+            ),
+            button: TextStyle(
+              fontSize: 18,
+            ),
+            headline5: TextStyle(
+              color: primaryColor,
+              fontSize: 10,
+            ),
+          )
+          .apply(
+            fontFamily: 'NotoSansJP',
+            fontSizeDelta: 2,
+          );
+  Widget _buildHomePage() {
+    return StreamBuilder<DocumentSnapshot>(
+      stream: FirebaseFirestore.instance
+          .collection(DbHandler.usersColletion)
+          .doc(FirebaseAuth.instance.currentUser?.email)
+          .snapshots(),
+      builder: (context, snapshot) {
+        if (!snapshot.hasData) {
+          return LinearProgressIndicator();
+        }
+        //TODO(hiroki): 強制アンラップのハンドリング
+        if (snapshot.data!.exists &&
+            snapshot.data?.get(FieldPath(['role'])) == 1) {
+          return HomeScreen();
+        } else if (snapshot.data?.get(FieldPath(['role'])) == 0) {
+          return ReasonsForCreatingTheSalonScreen();
+        }
```

Comment:
why use null safety on snapshot.data ? It will be non-null here because of L128

Reply:
I changed to non null safe.